src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -77,8 +88,16 @@ private void initResource() {
 
     @Test
     public void testCreateDataset() throws Exception {
+        DataSet result = new DataSet();
+        when(datasetService.getDatasetByName("test")).thenReturn(null);
+        when(datasetService.createDataset(any(), any())).thenReturn(result);
+        when(uriInfo.getRequestUriBuilder()).thenReturn(uriBuilder);
+        when(uriBuilder.replacePath(anyString())).thenReturn(uriBuilder);
+        when(uriBuilder.build(anyString())).thenReturn(new URI("/api/dataset/1"));
+
         initResource();
-        Response response = resource.createDataset(authUser, "");
+        Response response = resource.createDataset(authUser, uriInfo, "{\"properties\":[{\"propertyName\":\"Dataset Name\",\"propertyValue\":\"test\"}]}");
+
         assertEquals(201,response.getStatus());
     }
```

Comment:
Now that we have increased error handling in the resource, this is a great time to add tests for that. Try adding failure case tests that look for a bad request exception when the dataset is not filled out correctly. I would consider adding one failure case for each of the different error cases to make sure each individual one is handled.